File: Week_01/70.爬楼梯.py
#
# @lc app=leetcode.cn id=70 lang=python3
#
# [70] 爬楼梯
#

# @lc code=start
class Solution:
    def climbStairs(self, n: int) -> int:
        if n < 3:
            return n
        a, b = 1, 2
        for i in range(2, n):
            temp = a + b
            a = b
            b = temp
        return b


# @lc code=end

File: Week_09/70.爬楼梯.cpp
/*
 * @lc app=leetcode.cn id=70 lang=cpp
 *
 * [70] 爬楼梯
 */

// @lc code=start
class Solution
{
public:
    int climbStairs(int n)
    {
        if (n < 3)
        {
            return n;
        }
        int dp[2]{1, 2}, tmp;
        for (int i = 3; i <= n; i++)
        {
            tmp = dp[0] + dp[1];
            dp[0] = dp[1];
            dp[1] = tmp;
        }
        return dp[1];
    }
};
// @lc code=end
